server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -339,7 +447,9 @@ public void onFailure(Exception e) {
             } else {
                 responseActionListener = listener;
             }
-            if (sendFullVersion || previousState.nodes().nodeExists(destination) == false) {
+            if (sendRemoteState && destination.isRemoteClusterStateEnabled() && destination.isRemoteRoutingTableEnabled()) {
```

Comment:
can you just checking RemotePublicationEnabled (as it inherently means state is enabled anyway) and should be verified at the state of the process thats it.

Reply:
RemotePublicationEnabled enabled means that publication is enabled at the cluster manager node. But we need to check the flags for the destination node as well to know it supports the remote publication. This is required for mixed cluster mode.

Reply:
In case of mixed mode as well, you just need to check RemotePublicationEnabled on destination node only

Reply:
added RemotePublicationEnabled in discovery node